test/fw/ptl/utils/pbs_logutils.py

```diff
@@ -538,6 +557,172 @@ def get_log_files(self, hostname, path, start, end, sudo=False):
+        :param logval: The entity requested, an instance of a
+                        Scheduler, Server or MoM object, or the
+                        string 'tracejob' for tracejob
+        :type logval: str or object
+        :returns Last ``<n>`` lines of logfile for ``Server``,
+                  ``Scheduler``, ``MoM or tracejob``
+        """
+
+        # if syslog utility is not rsyslogd, throw error
+        try:
+            cmd_sys = "ps -e | grep syslog | awk '{print $NF}'"
+            utility = self.du.run_cmd(hosts=hostname, cmd=cmd_sys,
+                                      as_script=True,
+                                      level=logging.DEBUG2)['out']
+
+            if utility[0] != "rsyslogd":
+                raise PbsConfigError(rc=1, rv=None,
+                                     msg="Currently only rsyslog is supported")
+
+            severity = self.du.run_cmd(hosts=hostname,
+                                       cmd="cat /etc/pbs.conf "
+                                           "| grep -w PBS_SYSLOGSEVR | "
+                                           "awk '{print substr($0,length,1)}'",
+                                       as_script=True,
+                                       level=logging.DEBUG2)['out'][0]
+
+            if severity is None:
+                raise PbsConfigError(rc=1, rv=None,
+                                     msg="Set PBS_SYSLOGSEVR in pbs.conf")
+
+            facility = self.du.run_cmd(hosts=hostname,
+                                       cmd="cat /etc/pbs.conf "
+                                           "| grep -w PBS_SYSLOG | "
+                                           "awk '{print substr($0,length,1)}'",
+                                       as_script=True,
+                                       level=logging.DEBUG2)['out'][0]
+
+            list_of_syslog_files = self._get_rsyslog_files(hostname=hostname,
+                                                           severity=severity,
+                                                           facility=facility)
+
+            lines = []
+            if n == "ALL":
+                n = 5000
+            if hostname is None:
+                hostname = socket.gethostname()
+
+            if logval is 'sched_logs':
+                daemon_str = 'pbs_sched'
+            elif logval is 'server_logs':
+                daemon_str = 'Server@'
+            elif logval is 'mom_logs':
+                daemon_str = 'pbs_mom'
+            elif logval is 'comm_logs':
+                daemon_str = 'pbs_comm'
+            else:
+                daemon_str = None
+
+            for x in list_of_syslog_files:
+                run_cmd = [self.du.which(hostname, 'cat',
+                                         level=logging.INFOCLI2), x,
+                           " | tail -", str(n)]
+                l_sys = self.du.run_cmd(hosts=hostname, cmd=run_cmd, sudo=True,
+                                        level=logging.DEBUG2)['out']
```

Comment:
FileUtils().tail was failing because some machines might not have tail / tail command cannot be used with sudo. In this case using du.run_cmd was easier than using du.cat()